test/y2storage/proposal/settings_generator/ng_test.rb

```diff
@@ -75,14 +79,17 @@ def volume_from_settings(settings, mount_point)
       next_settings = subject.next_settings
 
       expect(next_settings).to be_a(Y2Storage::ProposalSettings)
-      expect(next_settings.object_id).to_not eq(settings.object_id)
+      expect(next_settings.to_s).to_not eq(settings.to_s)
     end
 
     context "when called for first time" do
       it "returns the same values as the initial settings" do
         next_settings = subject.next_settings
 
-        expect(next_settings.to_s).to eq(settings.to_s)
+        settings_content = remove_object_ids(settings.to_s)
+        next_settings_content = remove_object_ids(next_settings.to_s)
+
+        expect(next_settings_content).to eq(settings_content)
```

Comment:
If this approach looks good to you, I could do the same with the added example to test the initial guided proposal at  https://github.com/yast/yast-storage-ng/pull/889/files#diff-024fcd1dc73eb2f382db64fe6cb7529fR264. But if it smells bad... I'll undo it :)

Reply:
Comparing string representations of objects does not seem to be the right way to me, I'd avoid that if possible. IMHO too fragile...

Reply:
The test was already using the string representation (I guess because of `ProposalSettings` is redefining the [`#to_s`](https://github.com/yast/yast-storage-ng/blob/06c72586a788e209bb04dc350a0a2ec66ba542b2/src/lib/y2storage/proposal_settings.rb#L257))

So, shall we change this?

Reply:
Changed. See https://github.com/yast/yast-storage-ng/pull/889/commits/7b3d778a20bef749c9bf7226dee40f3c77a445f4 and https://github.com/yast/yast-storage-ng/pull/889/commits/a75490ebf2f42e5ec565e888f440a8ca9aeda8a7.